### ISSUE TITLE
Count dynamic block exceed...() statistics by Netmask rather than individual IP address

### ISSUE THREAD

**Issue by neilcook** [PowerDNS/pdns#4993]
- Program: dnsdist 
 - Issue type: Feature request

### Short description
The exceedXXX() functions used for triggering dynamic blocks return a set of ComboAddresses containing the IP addresses that matched the rule. However they only count based on individual IP addresses, (i.e. /32 for v4 and /128 for v6). It would be nice if they had the capability to count based on a netmask, e.g. exceedServFails(rate, seconds, v4mask, v6mask)

An example would be: exceedServFailsMask(20, 10, 32, 64)

Which would count v4 addresses by individual IP, but v6 addresses by /64 aggregation. 

The return value from these functions should probably be a set of Netmasks or a NetmaskGroup rather than a set of ComboAddress's, otherwise the set of ComboAddress's returned would be potentially huge.

### Usecase
Count groups of IP addresses rather than individual IPs addresses when creating dynamic blocks.

### Description
See above.

**Comment by rgacogne** [comment]
I don't think we will implement this for the `exceed*` functions that are pretty much deprecated at this point, but I would like to support this in `DynBlockRulesGroup`.
